posthog/clickhouse/migrations/0175_materialize_ai_is_error.py

```diff
@@ -0,0 +1,39 @@
+from posthog.clickhouse.client.connection import NodeRole
+from posthog.clickhouse.client.migration_tools import run_sql_with_exceptions
+
+ADD_COLUMN_SHARDED_EVENTS = """
+ALTER TABLE sharded_events
+ADD COLUMN IF NOT EXISTS `mat_$ai_is_error` Nullable(String)
+MATERIALIZED JSONExtract(properties, '$ai_is_error', 'Nullable(String)')
+"""
+
+ADD_COLUMN_EVENTS = """
+ALTER TABLE events
+ADD COLUMN IF NOT EXISTS `mat_$ai_is_error` Nullable(String)
+COMMENT 'column_materializer::properties::$ai_is_error'
+"""
+
+ADD_BLOOM_FILTER_INDEX_SHARDED_EVENTS = """
```

Comment:
if a number of possible values for `$ai_is_error` property is small (let's say it's boolean so values would be: true, false, null, 1,0), it would be much faster to use **set** skip index:
- https://clickhouse.com/docs/optimize/skipping-indexes#set
- https://clickhouse.com/docs/optimize/skipping-indexes/examples

```sql
ALTER TABLE sharded_events
ADD INDEX IF NOT EXISTS `set_$ai_is_error` `mat_$ai_is_error`
TYPE set(7)
```